src/events/balancer.py

```diff
@@ -195,16 +196,19 @@ def rebalance(self, event: ActionEvent) -> None:
                 not self.balancer_manager.cruise_control.ready,
                 "CruiseControl balancer service has not yet collected enough data to provide a partition reallocation proposal",
             ),
+            (
```

Comment:
**todo:** Can we add a conditional for asserting:

```python
if event.params.get("brokerid") not in [broker.id for broker in self.charm.state.brokers]:
    event.fail("invalid brokerid")  # or similar
```

Reply:
So I definitely noticed the Action failing for unknown reasons during development. I think we should fail the action early with meaningful messages, as non-200 isn't informative enough. We **should** also fail for a 200, if that isn't there already.

Reply:
I apologize, could you clarify this last statement, please?

> We should also fail for a 200, if that isn't there already.

Reply:
I don't think we handle a failed `client.post()` raising ApiError. I'm blind, and forgot we do handle `!= 200`, so ignore that point, sorry.

Reply:
Check added in 4a9ff786cdca